src/components/schedule-page/new-month-page.component.tsx

```diff
@@ -5,33 +5,36 @@ import React, { ChangeEvent, useEffect, useRef } from "react";
 import { useDispatch, useSelector } from "react-redux";
 import { TranslationHelper } from "../../helpers/translations.helper";
 import { ApplicationStateModel } from "../../state/models/application-state.model";
-import { createMonthKey } from "../../state/reducers/history.reducer";
 import { getDateWithMonthOffset } from "../../state/reducers/month-state/schedule-data/common-reducers";
 import { ScheduleDataActionCreator } from "../../state/reducers/month-state/schedule-data/schedule-data.action-creator";
 import { Button } from "../common-components";
 import { useScheduleConverter } from "./import-buttons/hooks/use-schedule-converter";
+import { ScheduleKey } from "../../api/persistance-store.model";
+import { MonthSwitchActionCreator } from "../../state/reducers/month-state/schedule-data/month-switch.action-creator";
 
 export function NewMonthPlanComponent(): JSX.Element {
   const state = useSelector((state: ApplicationStateModel) => state);
   const currentMonth = state.actualState.persistentSchedule.present.schedule_info.month_number;
   const currentYear = state.actualState.persistentSchedule.present.schedule_info.year;
   const nextDate = getDateWithMonthOffset(currentMonth, currentYear, 1);
   const prevDate = getDateWithMonthOffset(currentMonth, currentYear, -1);
-  const nextMonth = state.history[createMonthKey(nextDate.getMonth(), nextDate.getFullYear())];
-  const prevMonth = state.history[createMonthKey(prevDate.getMonth(), prevDate.getFullYear())];
-  const { scheduleModel, setSrcFile } = useScheduleConverter();
+  const nextMonth =
+    state.history[new ScheduleKey(nextDate.getMonth(), nextDate.getFullYear()).dbKey];
+  const prevMonth =
+    state.history[new ScheduleKey(prevDate.getMonth(), prevDate.getFullYear()).dbKey];
+  const { monthModel, setSrcFile } = useScheduleConverter();
 
-  const hasNext =
-    nextMonth !== undefined && nextMonth.persistentSchedule.present.month_info.dates.length > 0;
-  const hasPrevious =
-    prevMonth !== undefined && prevMonth.persistentSchedule.present.month_info.dates.length > 0;
+  const hasValidNext =
+    nextMonth !== undefined && nextMonth.month_info.dates.length > 0 && !nextMonth.isAutoGenerated;
```

Comment:
To blokuje nam wyświetlanie możliwość kopiowania z miesiąca, który jest wygenerowany